auth0/structure_auth0_connection.go

```diff
@@ -8,317 +8,372 @@ import (
 
-	switch o := options.(type) {
+	switch connectionOptions := options.(type) {
 	case *management.ConnectionOptions:
-		m = flattenConnectionOptionsAuth0(d, o)
+		m = flattenConnectionOptionsAuth0(d, connectionOptions)
 	case *management.ConnectionOptionsGoogleOAuth2:
-		m = flattenConnectionOptionsGoogleOAuth2(o)
+		m = flattenConnectionOptionsGoogleOAuth2(connectionOptions)
 	case *management.ConnectionOptionsGoogleApps:
-		m = flattenConnectionOptionsGoogleApps(o)
+		m = flattenConnectionOptionsGoogleApps(connectionOptions)
 	case *management.ConnectionOptionsOAuth2:
-		m = flattenConnectionOptionsOAuth2(o)
+		m = flattenConnectionOptionsOAuth2(connectionOptions)
 	case *management.ConnectionOptionsFacebook:
-		m = flattenConnectionOptionsFacebook(o)
+		m = flattenConnectionOptionsFacebook(connectionOptions)
 	case *management.ConnectionOptionsApple:
-		m = flattenConnectionOptionsApple(o)
+		m = flattenConnectionOptionsApple(connectionOptions)
 	case *management.ConnectionOptionsLinkedin:
-		m = flattenConnectionOptionsLinkedin(o)
+		m = flattenConnectionOptionsLinkedin(connectionOptions)
 	case *management.ConnectionOptionsGitHub:
-		m = flattenConnectionOptionsGitHub(o)
+		m = flattenConnectionOptionsGitHub(connectionOptions)
 	case *management.ConnectionOptionsWindowsLive:
-		m = flattenConnectionOptionsWindowsLive(o)
+		m = flattenConnectionOptionsWindowsLive(connectionOptions)
 	case *management.ConnectionOptionsSalesforce:
-		m = flattenConnectionOptionsSalesforce(o)
+		m = flattenConnectionOptionsSalesforce(connectionOptions)
 	case *management.ConnectionOptionsEmail:
-		m = flattenConnectionOptionsEmail(o)
+		m = flattenConnectionOptionsEmail(connectionOptions)
 	case *management.ConnectionOptionsSMS:
-		m = flattenConnectionOptionsSMS(o)
+		m = flattenConnectionOptionsSMS(connectionOptions)
 	case *management.ConnectionOptionsOIDC:
-		m = flattenConnectionOptionsOIDC(o)
+		m = flattenConnectionOptionsOIDC(connectionOptions)
 	case *management.ConnectionOptionsAD:
-		m = flattenConnectionOptionsAD(o)
+		m = flattenConnectionOptionsAD(connectionOptions)
 	case *management.ConnectionOptionsAzureAD:
-		m = flattenConnectionOptionsAzureAD(o)
+		m = flattenConnectionOptionsAzureAD(connectionOptions)
 	case *management.ConnectionOptionsADFS:
-		m = flattenConnectionOptionsADFS(o)
+		m = flattenConnectionOptionsADFS(connectionOptions)
 	case *management.ConnectionOptionsSAML:
-		m = flattenConnectionOptionsSAML(o)
+		m = flattenConnectionOptionsSAML(connectionOptions)
 	}
 
 	return []interface{}{m}
 }
 
-func flattenConnectionOptionsGitHub(o *management.ConnectionOptionsGitHub) interface{} {
-	return map[string]interface{}{
-		"client_id":                o.GetClientID(),
-		"client_secret":            o.GetClientSecret(),
-		"set_user_root_attributes": o.GetSetUserAttributes(),
-		"non_persistent_attrs":     o.GetNonPersistentAttrs(),
-		"scopes":                   o.Scopes(),
-	}
+func optionsBlockIsNotDefined(d ResourceData) bool {
```

Comment:
This now allows to have a connection without options block defined in the terraform config.